sys/include/fs/fatfs.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2017 HAW-Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_fatfs  FatFs integration
+ * @ingroup     sys_fs
+ * @{
+ *
+ * @file
+ * @brief       FatFs integration for vfs
+ *
+ * @author      Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ */
+
+#ifndef FATFS_H
+#define FATFS_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef VFS_DIR_BUFFER_SIZE
+#define VFS_DIR_BUFFER_SIZE (44)
+#endif
+
+#ifndef VFS_FILE_BUFFER_SIZE
+#define VFS_FILE_BUFFER_SIZE (72)
+#endif
+
+#include "fatfs/ff.h"
+#include "vfs.h"
+
+#ifndef FATFS_YEAR_OFFSET
+#define FATFS_YEAR_OFFSET (1980)
+#endif
+
+#define EPOCH_YEAR_OFFSET (1970)
+
+/** Size of the buffer needed for directory -> should be: sizeof(DIR)*/
+#define FATFS_DIR_SIZE (44)
+/** the problem with the above is: it's not possible to use sizeof(DIR) as this is later usen in #if (see below) */
+
+/** Size of the buffer needed for directory -> should be: sizeof(fatfs_file_desc_t)*/
+#define FATFS_FILE_SIZE (72)
+
+#define FATFS_MAX_VOL_STR_LEN (5)   /**< size needed for volume strings like "n:/" where n is the volume id */
```

Comment:
I didn't look how the macro is used, but `sizeof("n:/") == 4`. Where does the extra character come from?